scripts/build/Jenkins/Jenkinsfile

```diff
@@ -38,15 +38,15 @@ def pipelineParameters = [
     booleanParam(defaultValue: false, description: 'Recreates the volume used for the workspace. The volume will be created out of a snapshot taken from main.', name: 'RECREATE_VOLUME')
 ]
 
-def palSh(cmd, lbl = '', winSlashReplacement = true) {
+def palSh(cmd, lbl = '', winCharReplacement = true) {
     if (env.IS_UNIX) {
         sh label: lbl,
            script: cmd
     } else {
-        if (winSlashReplacement) {
+        if (winCharReplacement) {
             cmd = cmd.replace('/','\\')
+            cmd = cmd.replace('%', '%%')
```

Comment:
these two replacements together could cause issues in the future
The `/` -> `\\` is because we build paths "unix-like" and then need to convert them in Windows. The calls on 279 and 281 are just lucky to not need it because we use the system python for the mount script
The `%` -> `%%` replacement is for URL cases. Which seems is just an issue for the calls to the mount script.

I would recommend to either:
a) do the `%` -> `%%` replacement for the mount script call; or
b) add a different parameter in this function for this replacement

Approving anyway since it solves the issue and this suggestion is just to make it easier for the future

Reply:
I should mention that both `/` and `%` were already being replaced together as of https://github.com/o3de/o3de/pull/6688. This change just makes the `%` -> `%%` replacement optional instead of mandatory.

To clarify: The `%` character is not actually being used in the mount script, but any usage of `palSh` that has URLs encoded with `%2` in order to escape the character in Windows.

I'll go with option b), since it's less impactful than hunting down each `palSh` call and determining if they need to the escape character for its situation.